### PR TITLE
fix: Enhance artifact visualization. Fixes #8619

### LINKED ISSUE #8619: The horizontal/vertical option on the graph panel does not persist
This option:

image

Does not persist if leave the page. It should.

### PR DESCRIPTION
**Pull request by alexec**
Signed-off-by: Alex Collins <alex_collins@intuit.com>

Fixes #8619

* Made artifact panel wider (50%) to make it easie to see large artifacts.
* Filter out artifacts from the DAG that cannot be visualized (only show files and folders).
* Update CSP to allow inline-styles, so pages can be styled nicer.
* Updated examples to make them less trivial.
* Fixed a bug where Git artifacts key ignore the branch.
* Fixed new and old bugs, annoyances, and out of date help in the log viewer.
* Fix bug in persistence of the DAGS horizontal/vertical toggle. #8619

### CODE REVIEW

File: examples/artifacts-workflowtemplate.yaml [hunk 1]
```diff
@@ -9,45 +9,83 @@ spec:
...
+          - name: setup
+            image: argoproj/argosay:v2
+            command:
+              - sh
+              - -c
+            args:
+              - |
+                mkdir -p /out/assets
+          - name: gnuplot
+            image: remuslazar/gnuplot
```
Comment (alexec) [inline examples/artifacts-workflowtemplate.yaml:39]: `gnuplot` can create some nice charts.

File: examples/ci-workflowtemplate.yaml [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+            subPath: GOMODCACHE
+          - mountPath: /root/.cache/go-build
+            name: work
+            subPath: GOCACHE
+        workingDir: /go/src/github.com/golang/example
+        command: [ sh, -euxc ]
+        args:
+          - |
+            go install github.com/jstemmer/go-junit-report@latest
+            go install github.com/alexec/junit2html@v0.0.2
```
Comment (alexec) [inline examples/ci-workflowtemplate.yaml:183]: I had to write this myself.

Comment (terrytangyuan) [reply examples/ci-workflowtemplate.yaml:183]: Can we put this in our examples folder as it's a simple script?

**Review by terrytangyuan — COMMENTED** [review]
LGTM. Just a small comment but not a blocker.

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**